docs/docs/guides/integrations/langchain.md

```diff
@@ -0,0 +1,239 @@
+---
+sidebar_position: 2
+hide_table_of_contents: true
+---
+
+# LangChain
+
+Weave is designed to make tracking and logging all calls made through the [LangChain Python library](https://github.com/langchain-ai/langchain) effortless.
+
+When working with LLMs, debugging is inevitable. Whether a model call fails, an output is misformatted, or nested model calls create confusion, pinpointing issues can be challenging. LangChain applications often consist of multiple steps and LLM call invocations, making it crucial to understand the inner workings of your chains and agents.
+
+Weave simplifies this process by automatically capturing traces for your [LangChain](https://python.langchain.com/v0.2/docs/introduction/) applications. This enables you to monitor and analyze your application's performance, making it easier to debug and optimize your LLM workflows.
+
+
+## Getting Started
+
+To get started, simply call `weave.init()` at the beginning of your script. The argument in weave.init() is a project name that will help you organize your traces.
+
+```python
+import weave
+from langchain_core.prompts import PromptTemplate
+from langchain_openai import ChatOpenAI
+
+# Initialize Weave with your project name
+# highlight-next-line
+weave.init("langchain_demo")
+
+llm = ChatOpenAI()
+prompt = PromptTemplate.from_template("1 + {number} = ")
+
+llm_chain = prompt | llm
+
+output = llm_chain.invoke({"number": 2})
+
+print(output)
+```
+
+## Traces
+
+Storing traces of LLM applications in a central database is crucial during both development and production. These traces are essential for debugging and improving your application by providing a valuable dataset.
+
+Weave automatically captures traces for your LangChain applications. It will track and log all calls made through the LangChain library, including prompt templates, chains, LLM calls, tools, and agent steps. You can view the traces in the Weave web interface.
+
+[![langchain_trace.png](imgs/langchain_trace.png)](https://wandb.ai/parambharat/langchain_demo/weave/calls)
+
+## Manually Tracing Calls
```

Comment:
I am not understanding why i would ever do these two options? Under what conditions is that needed? As a user, I am thinking... why does the standard weave.init not work here?

Reply:
:thinking: Actually, since we patch in the Tracer during `weave.init`, these two patterns are unnecessary. However, it's there for feature parity with other tracers and handlers in LangChain that have these two patterns to allow users to add tracers and callbacks manually.

Reply:
I think the other manual tracers don't have this auto-patch magic though. It just seems like this is extra and in practice i am not seeing a situation where the user would do this.

Reply:
Makes sense. I'm removing these from the docs and the code.

Reply:
You probably don't need to change any code, right?

Reply:
Actually I just remembered why they are also there. Here's a use case. The user conditionally wants to enable weave tracing for part of their code. 

The workflow might look something like this

```
import os

os.environ["WEAVE_TRACE_LANGCHAIN"] = "false" # <- explicitly disable global tracing.

from weave.integrations.langchain import weave_tracing_enabled
from langchain_core.prompts import PromptTemplate
from langchain_openai import ChatOpenAI
import weave

# Initialize Weave with your project name
# highlight-next-line
weave.init(
    "langchain_demo"
)  # <-- we don't enable tracing here because the env var is explicitly set to `false`

llm = ChatOpenAI()
prompt = PromptTemplate.from_template("1 + {number} = ")

llm_chain = prompt | llm

# highlight-next-line
with weave_tracing_enabled():  # <-- this enables tracing only for this chain invoke.
    output = llm_chain.invoke({"number": 2})


llm_chain.invoke({"number": 4})  # <-- this will not have tracing enabled for langchain calls - but openai calls get traced

```

Reply:
I see - ok. Do you mind clarifying that then in the code docs & markdown so we & users are clear when to do this?

Reply:
Updated the docs
1. Added a note in the manual tracing section and added comments to the code.